torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -0,0 +1,337 @@
+
+template <typename T>
+size_t bufferSize(T t) {
+  size_t size = 1;
+  for (int i = 0; i < t.ndim(); i++) {
+    size *= t.dim(i).template AsNode<IntImm>()->value();
+  }
+  return size;
+}
+
+struct TensorExprKernel {
+  std::vector<Buffer*> buffer_args;
+  Tensor* tensor_output;
+  std::unordered_map<int64_t, Buffer> buffers;
+  std::unordered_map<int64_t, Tensor> tensors;
+  Stmt stmt;
+
+  explicit TensorExprKernel(const Node* node) {
+    auto subgraph = node->g(attr::Subgraph);
+
+    // Bind inputs to buffers.
+    auto inputs = subgraph->inputs();
+    for (auto const& input : subgraph->inputs()) {
+      buffers.emplace(input->unique(), texprBuffer(input));
+      buffer_args.push_back(&buffers.at(input->unique()));
+    }
+
+    // Bind nodes to tensor compute expressions.
+    std::unordered_map<int64_t, Expr> constants;
+    for (auto const& n : subgraph->nodes()) {
+      if (n->kind() == prim::Constant) {
+        const auto val = toIValue(n->output()).value();
+        if (val.isDouble()) {
+          constants[n->output()->unique()] = FloatImm::make(val.toDouble());
+        } else if (val.isInt()) {
+          constants[n->output()->unique()] = IntImm::make(val.toInt());
+        } else {
+          LOG(FATAL) << "Unhandled constant datatype";
+        }
+        continue;
+      }
+
+      if (n->kind() == aten::add) {
+        auto* lhs = n->inputs()[0];
+        auto* rhs = n->inputs()[1];
+        auto luniq = lhs->unique();
+        auto runiq = rhs->unique();
+
+        if (tensors.count(luniq) && tensors.count(runiq)) {
+          std::cerr << "aten::add(T, T)\n";
+          auto tt = n->output()->type()->cast<TensorType>();
+          auto exprDims = texprSizes(tt->sizes());
+          std::vector<DimArg> dims(exprDims.begin(), exprDims.end());
+
+          auto lt = tensors.at(luniq);
+          auto rt = tensors.at(runiq);
+          tensors.emplace(
+              n->output()->unique(),
+              Compute(
+                  "aten__add", dims, [lt, rt](const std::vector<Var>& axes) {
+                    return lt(axes[0]) + rt(axes[0]);
+                  }));
+        } else if (tensors.count(luniq) && buffers.count(runiq)) {
+          LOG(FATAL) << "Unhandle aten::add(Tensor, Buffer)";
+        } else if (buffers.count(luniq) && tensors.count(runiq)) {
+          LOG(FATAL) << "Unhandle aten::add(Buffer, Tensor)";
+        } else if (buffers.count(luniq) && buffers.count(runiq)) {
+          std::cerr << "aten::add(B, B)\n";
+          auto tt = n->output()->type()->cast<TensorType>();
+          auto exprDims = texprSizes(tt->sizes());
+          std::vector<DimArg> dims(exprDims.begin(), exprDims.end());
+
+          auto lt = buffers.at(luniq);
+          auto rt = buffers.at(runiq);
+          tensors.emplace(
+              n->output()->unique(),
+              Compute(
+                  "aten__add", dims, [lt, rt](const std::vector<Var>& axes) {
+                    return lt(axes[0]) + rt(axes[0]);
```

Comment:
If you only want to use the 1-d version of the Compute, there is a 1D, 2D overload of the compute. 

https://github.com/bertmaher/pytorch/blob/pytorch_fusion/torch/csrc/jit/tensorexpr/tests/expr_test.cpp#L55